### PR TITLE
Remove the sleep call within the APIRefresher

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR removes the timeout within the APIRefresher thread.

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/lightning_app/core/api.py [hunk 1]
```diff
@@ -70,21 +69,18 @@ class SessionMiddleware:
 
 
 class UIRefresher(Thread):
-    def __init__(self, api_publish_state_queue, api_response_queue, refresh_interval: float = 0.1) -> None:
```
Comment (awaelchli) [inline src/lightning_app/core/api.py:73]: This is not a timeout.

The PR does not explain why we remove this.

File: src/lightning_app/core/api.py [hunk 2]
```diff
@@ -70,21 +69,18 @@ class SessionMiddleware:
...
-        self.refresh_interval = refresh_interval
 
     def run(self):
         # TODO: Create multiple threads to handle the background logic
         # TODO: Investigate the use of `parallel=True`
         try:
             while not self._exit_event.is_set():
                 self.run_once()
-                # Note: Sleep to reduce queue calls.
-                sleep(self.refresh_interval)
```
Comment (awaelchli) [inline src/lightning_app/core/api.py:87]: How will we address the increase in queue calls?

File: src/lightning_app/core/api.py (hunk 1 above)
Comment (tchaton) [reply src/lightning_app/core/api.py:73]: Rephrase to: `Remove the sleep call within the APIRefresher`

File: src/lightning_app/core/api.py (hunk 2 above)
Comment (tchaton) [reply src/lightning_app/core/api.py:87]: Answer in private.

Comment (Borda) [reply src/lightning_app/core/api.py:87]: @awaelchli are you fine with it? :)

**Review by awaelchli — APPROVED** [review]

**Pull request closed without merging**